ui/app/AppLayouts/Browser/panels/FavoritesBar.qml

```diff
@@ -35,18 +35,28 @@ RowLayout {
                 acceptedButtons: Qt.LeftButton | Qt.RightButton
                 z: 51
                 onClicked: function (mouse) {
+                    const isAddBookmarkButton = url === " "
```

Comment:
I don't remember how that works. Where does `url` come from and how come it is a space when it's the bookmark button?

Reply:
Yeah, it is quite weird.... It took me a bit of time to understand. Here is how I think it works: The add bookmark button is a bookmark saved with the url set at empty. So when we display the bookmark in the toolbar we have a special case for the empty space which correspond to the button to add a bookmark

Reply:
Yea I did that part of adding a new item to the model with " " string. This in turn represents the "add fav button". I did this to handle the requirement that the addFav button should always be at the end after a bookmark is added/deleted.

Reply:
Ah ok, makes sense. What I would recommend then is to replace the space with a constant. Something like `NEW_BOOKMARK = "ADD_NEW_BOOKMARK"`
That way, it's clearer for us, and also is easier to expend if we need to send it from multiple places.

Reply:
np, I will do the constant, but that will be a breaking change to rename it to Add new bookmark right? cause people will start to see an empty bookmark. @jrainville ok with that?

Reply:
Oh crap, I misunderstood the behavior.
Ok then the constant is still a good idea, but make the constant equal the space. So `NEW_BOOKMARK = " "`

Reply:
Updated!

Reply:
I think you just need to also update the `addBookmark(" ")` and `deleteBookmark(" ")` calls throughout the code to use the Contsant

Reply:
you're right, all updated this time (I think)